source/parser/x86_64/classifiers.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#pragma once
+
+#include <utility>
+
+#include "Type.h"
+#include "register_class.hpp"
+
+namespace smeagle::x86_64 {
+
+  struct Class {
+    RegisterClass lo, hi;
+    int pointer_indirections;
+    std::string name;
+  };
+
+  namespace st = Dyninst::SymtabAPI;
+
+  inline Class classify(st::typeScalar *t, int ptr_cnt) {
```

Comment:
Hmm maybe call this a RegisterClass to not confuse with what we normally think of as a class?

Reply:
oh wait a Class has a RegisterClass... ohno.  What about if we just reverse it? As long as the Class is internal it won't be as weird to instantiate something called just Class?

```cpp
struct RegisterClass {
    Class lo, hi;
}
```
or even just

```cpp
struct RegisterClass {
    ClassName lo, hi;
}
```
or 

```cpp
struct RegisterClasses {
    RegisterClass lo, hi;
}
```
Maybe it's not a big deal- what are your thoughts?

Reply:
I'm 100% with you on this one. This naming scheme was lifted whole-cloth from the LLVM implementation. Maybe something like `classification` would be a good return type from a function called `classify`?

Reply:
yes!